### DISCUSSION TITLE
Problem to navigate when I put my three routes inside a new router component.

### DISCUSSION THREAD

**Discussion by vagnerwentz** [react-navigation/react-navigation#10448] [Q&A] [answered]
### Versions of navigation
 "@react-navigation/native": "^6.0.8",
 "@react-navigation/native-stack": "^6.4.1",

I created a new router to uncouple to another files, so I did this route below and I imported this `PersonalAccountSignUp` into a `Router.tsx` file and when I try to navigate to `PersonalInformation`, this error appears

The action 'NAVIGATE' with payload {"name": "PersonalInformation"} was not handled by any navigator.
Do you have a screen named 'PersonalInformation'

### src/screens/SignUp/PersonalAccount/index.tsx
```
import { Stack } from '../../../Router'

import i18n from 'i18n-js'

import { PersonalInformation } from 'screens/SignUp/PersonalAccount/PersonalInformation'
import { PersonalLocationAndOccupation } from 'screens/SignUp/PersonalAccount/PersonalLocationAndOccupation'
import { OccupationList } from 'screens/SignUp/PersonalAccount/PersonalLocationAndOccupation/components/OccupationList'

import { CreatePersonalAccountContextProvider } from 'store/CreatePersonalAccount'

export function PersonalAccountSignUp() {
  return (
    <CreatePersonalAccountContextProvider>
      <Stack.Screen
        name="PersonalInformation"
        component={PersonalInformation}
        options={{
          title: i18n.t('registerPersonal.screenName')
        }}
      />
      <Stack.Screen
        name="PersonalLocationAndOccupation"
        component={PersonalLocationAndOccupation}
        options={{
          title: i18n.t('registerPersonal.screenName')
        }}
      />
      <Stack.Screen
        name="OccupationList"
        component={OccupationList}
        options={{
          title: 'Occupation or profession'
        }}
      />
    </CreatePersonalAccountContextProvider>
  )
}
```

### src/Router.tsx
```
...
<Stack.Group>
    <Stack.Screen
        name="PersonalAccountSignUp"
        component={PersonalAccountSignUp}
    />
</Stack.Group>
...
```

**Comment by raajnadar** [comment, marked as answer]
Can you please add a snack repro?

**Reply by vagnerwentz** [reply]
I can try, because this a private repository.

**Reply by vagnerwentz** [reply]
What do you think what is very useful to see? @raajnadar

**Reply by raajnadar** [reply]
The stack navigator with 3 routes and how you are navigating to the screen in https://snack.expo.dev/

**Reply by vagnerwentz** [reply]
Here @raajnadar https://snack.expo.dev/aFYKakKCe

**Reply by raajnadar** [reply]
The code is not working, please provide a runnable code.

**Reply by vagnerwentz** [reply]
I don't know how to handle inside the snack expo

**Reply by raajnadar** [reply]
Add a small repro code on Github repository

**Reply by vagnerwentz** [reply]
I want that when I press the button to create personal account, will go to PersonalInformation route
@raajnadar https://github.com/vagnerwentz/super-duper

To run you need to pull the repository and runs yarn command and start the project with expo r -c

**Reply by vagnerwentz** [reply]
I got something, I put a Stack.Navigator around my PersonalAccountSignUp, but nows the header broke =/

**Reply by vagnerwentz** [reply]
@raajnadar Hey, did you find something?

**Reply by raajnadar** [reply]
Yes, the direct parent of a `Stack.Screen` should be either `Stack.Navigator` or `Stack.Group` for `Stack.Group` it should always be a `Stack.Navigator`

I thought the same when seeing the above code, Can you send the screenshot with the `Stack.Navigator` I need to check what is broken in the Header.

You need `Stack.Navigator` inside your `<CreatePersonalAccountContextProvider />` wrapper

**Reply by vagnerwentz** [reply]
I got something, but I needed to insert the same header inside the Stack.Navigator at PersonalAccount and at Router.tsx I needed to hide the header into the PersonalAccount.

PersonalAccount.tsx
```
export function PersonalAccountSignUp() {
  return (
    <CreatePersonalAccountContextProvider>
      <Stack.Navigator
        screenOptions={{
          header: (props) => {
            return <Header {...props} />
          },
          headerTintColor: '#000',
          headerBackTitleVisible: false
        }}
      >
        <Stack.Screen
          name="PersonalInformation"
          component={PersonalInformation}
          options={{
            title: i18n.t('registerPersonal.screenName')
          }}
        />
        <Stack.Screen
          name="PersonalLocationAndOccupation"
          component={PersonalLocationAndOccupation}
          options={{
            title: i18n.t('registerPersonal.screenName')
          }}
        />
        <Stack.Screen
          name="OccupationList"
          component={OccupationList}
          options={{
            title: 'Occupation or profession'
          }}
        />
      </Stack.Navigator>
    </CreatePersonalAccountContextProvider>
  )
}
```

Router.tsx
```
<Stack.Navigator
      screenOptions={{
        header: (props) => {
          return <Header {...props} />
        },
        headerTintColor: '#000',
        headerBackTitleVisible: false
      }}
    >
...   
      <Stack.Group>
        <Stack.Screen
          name="CreateAccount"
          component={ChooseTypeAccount}
          options={{
            title: i18n.t('createAccount.screenName')
          }}
        />
        <Stack.Screen
          name="PersonalAccountSignUp"
          component={PersonalAccountSignUp}
          options={{
            headerShown: false
          }}
        />
      </Stack.Group>
...
    </Stack.Navigator>
```

**Reply by raajnadar** [reply]
I would suggest using the header from the child so that the back button works correctly & hide the header on parent navigator.

**Comment by satya164** [comment]
You can't just move screen configuration to a different file like that. If you need multiple navigators, use nesting: https://reactnavigation.org/docs/nesting-navigators/

But it's not recommended to nest multiple stack navigators. It's better to define all of them in a single navigator in a single file.

**Reply by vagnerwentz** [reply]
Well, but is it correct? Because think about at a big application
